sample-app-3/src/main/java/com/yoti/mobile/android/sdk/sampleapp/MainActivity.java

```diff
@@ -32,62 +40,108 @@
     private TextView mSdkEntry;
     private TextView mButtonTextEntry;
     private TextView mUseCaseEntry;
+    private ViewFlipper mSdkButtonViewFlipper;
 
+    @SuppressLint("NonConstantResourceId")
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         setContentView(R.layout.activity_main);
         YotiSDK.enableSDKLogging(true);
 
-        mYotiSDKButton = findViewById(R.id.button);
         mStatusContainer = findViewById(R.id.resultContainer);
         mStatusHeader = findViewById(R.id.resultHeader);
         mStatusMessage = findViewById(R.id.resultStatus);
         mScenarioEntry = findViewById(R.id.scenarioIdText);
         mSdkEntry = findViewById(R.id.sdkIdText);
         mButtonTextEntry = findViewById(R.id.buttonLabelText);
         mUseCaseEntry = findViewById(R.id.useCaseIdText);
-        createScenario();
 
         mSdkEntry.addTextChangedListener(scenarioUpdateListener);
         mScenarioEntry.addTextChangedListener(scenarioUpdateListener);
         mUseCaseEntry.addTextChangedListener(scenarioUpdateListener);
         mButtonTextEntry.addTextChangedListener(buttonTextListener);
 
-        mYotiSDKButton.setOnYotiButtonClickListener(new YotiSDKButton.OnYotiButtonClickListener() {
+        mSdkButtonViewFlipper = findViewById(R.id.sdkButtonViewFlipper);
+        handleSDKButtonThemeSelection(PARTNERSHIP_SDK_BUTTON);
+
+        RadioGroup buttonSelectionRadioGroup = findViewById(R.id.buttonSelectionRadioGroup);
+        buttonSelectionRadioGroup.setOnCheckedChangeListener((group, checkedId) -> {
+            switch (checkedId) {
+                case R.id.yotiRadioButton:
+                    handleSDKButtonThemeSelection(YOTI_SDK_BUTTON);
+                break;
+                case R.id.easyIdRadioButton:
+                    handleSDKButtonThemeSelection(EASY_ID_SDK_BUTTON);
+                    break;
+                default:
+                    handleSDKButtonThemeSelection(PARTNERSHIP_SDK_BUTTON);
+                    break;
+            }
+        });
+    }
+
+    private void handleSDKButtonThemeSelection(int selection) {
+        switch (selection) {
+            case YOTI_SDK_BUTTON:
+                YotiSDKButton yotiSdkButton = findViewById(R.id.yotiSDKButton);
+                mSdkButtonViewFlipper.setDisplayedChild(YOTI_SDK_BUTTON);
```

Comment:
Since it's a sample app we kept `ViewFlipper` and since we have multiple buttons we needs to take care of visibility of each button and bit difficult to maintain.

Reply:
Well, it is in the sample app, but also in `YotiButtonContainer` 😢

Reply:
In `YotiButtonContainer` we used `ViewFlipper` to avoid `setVisibility` everywhere as we got more themes, which will be bit more error prone☹️